docgen/src/guide/Custom_backend.md

```diff
@@ -0,0 +1,172 @@
+```
+
+Once your new route is ready, we get back to the frontend and are going to create a search client able to communicate with this server.
+
+## On the frontend: call your new backend routes
+
+### Searching for results
+
+A search client is an object which implements the method `search()`, called every time the user searches and refines results.
+
+```javascript
+search: (requests: SearchRequest[]) => Promise<{ results: SearchResponse[] }>
+```
+
+Since our server accepts the InstantSearch format as an input, we will only need to pass these requests to our backend in this method, and return the response.
+
+```javascript
+const customSearchClient = {
+  search(requests) {
+    return fetch('http://localhost:3000/search', {
+      method: 'post',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ requests }),
+    }).then(res => res.json());
+  }
+};
+```
+
+We use the [Fetch API](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API) in this example to query the server; make sure to take note of the [browser compatibility](https://developer.mozilla.org/en-US/docs/Web/API/Fetch_API#Browser_compatibility) before using it in production.
+
+> If you want to transform the data to be passed to your server, you can learn more about all the parameters that the `search()` method supports in the [Algolia API reference](https://www.algolia.com/doc/api-reference/api-parameters/?language=javascript#parameters-list).
+
+### Searching for facet values
+
+If your server exposes a *Search For Facet Values* endpoint, you can implement this feature in a similar way as the `search()` method with the `searchForFacetValues()` method.
+
+```javascript
+searchForFacetValues?: (requests: SearchForFacetValuesRequest[]) => Promise<{ facetHits: SearchForFacetValuesResponse[] }>
+```
+
+The implementation is the same as the `search()` method, except that we target `POST /sffv`.
+
+```javascript
+const customSearchClient = {
+  search(requests) {
+    /* ... */
+  },
+  searchForFacetValues(requests) {
+    return fetch('http://localhost:3000/sffv', {
+      method: 'post',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify({ requests }),
+    }).then(res => res.json());
+  }
+};
+```
+
+You’ll be able to search within your refinement lists from your custom backend.
+
+> Note that you will need to set the option [`searchForFacetValues`](https://community.algolia.com/instantsearch.js/v2/widgets/refinementList.html#struct-RefinementListWidgetOptions-searchForFacetValues) to `true` in your refinement list to make it searchable.
+
+## Using the search client with InstantSearch
+
+Now, we need to tell InstantSearch to use the search client that we’ve created. This is possible with the `searchClient` option. This parameter will disable all Algolia requests coming from the frontend and will proxy them to your own backend implementation.
+
+```javascript
+const App = () =>
```

Comment:
Add some parens around the body of the function.

Reply:
That's what I would have done too. Although I copy-pasted the code snippet from the documentation (which I can't find now).

Anyway, I'm for parens, will update.